client/matchmaking/admin-matchmaking-times.tsx

```diff
@@ -0,0 +1,690 @@
+
+function getCurrentMatchmakingTime(
+  type: MatchmakingType,
+  spec: RequestHandlingSpec<MatchmakingTimeJson | undefined>,
+): ThunkAction {
+  return abortableThunk(spec, async () => {
+    return await fetchJson<MatchmakingTimeJson | undefined>(
+      apiUrl`matchmaking-times/${encodeURIComponent(type)}/current`,
+      {
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function getMatchmakingTimesFuture(
+  type: MatchmakingType,
+  offset: number,
+  spec: RequestHandlingSpec<GetFutureMatchmakingTimesResponse>,
+): ThunkAction {
+  return abortableThunk(spec, async () => {
+    return await fetchJson<GetFutureMatchmakingTimesResponse>(
+      apiUrl`matchmaking-times/${encodeURIComponent(type)}/future?offset=${offset}`,
+      {
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function getMatchmakingTimesPast(
+  type: MatchmakingType,
+  offset: number,
+  spec: RequestHandlingSpec<GetPastMatchmakingTimesResponse>,
+): ThunkAction {
+  return abortableThunk(spec, async () => {
+    return await fetchJson<GetPastMatchmakingTimesResponse>(
+      apiUrl`matchmaking-times/${encodeURIComponent(type)}/past?offset=${offset}`,
+      {
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function addMatchmakingTime(
+  type: MatchmakingType,
+  startDate: number,
+  enabled: boolean,
+  spec: RequestHandlingSpec<MatchmakingTimeJson>,
+): ThunkAction {
+  return abortableThunk(spec, async dispatch => {
+    return await fetchJson<MatchmakingTimeJson>(
+      `/api/1/matchmaking-times/${encodeURIComponent(type)}`,
+      {
+        method: 'post',
+        body: encodeBodyAsParams<AddMatchmakingTimeRequest>({ startDate, enabled }),
+        signal: spec.signal,
+      },
+    )
+  })
+}
+
+function deleteMatchmakingTime(
+  type: MatchmakingType,
+  id: number,
+  spec: RequestHandlingSpec,
+): ThunkAction {
+  return abortableThunk(spec, async dispatch => {
+    await fetchJson(`/api/1/matchmaking-times/${encodeURIComponent(id)}`, {
+      method: 'delete',
+      signal: spec.signal,
+    })
+  })
+}
+
+const Container = styled(CenteredContentContainer)`
+  padding-top: 16px;
+  padding-bottom: 8px;
+`
+
+const PageHeadline = styled.div`
+  ${titleLarge};
+  margin-top: 16px;
+  margin-bottom: 8px;
+`
+
+const ErrorText = styled.div`
+  ${bodyLarge};
+  color: var(--theme-error);
+`
+
+const AddNewCard = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+  padding: 16px;
+  background-color: rgba(255, 255, 255, 0.03);
+  border-radius: 8px;
+  margin-bottom: 16px;
+`
+
+const DateInputContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 16px;
+`
+
+const DateInput = styled.input`
+  color: var(--theme-on-surface);
+  padding: 8px 12px;
+  border-radius: 4px;
+  border: 1px solid rgba(255, 255, 255, 0.1);
+  background-color: rgba(255, 255, 255, 0.05);
+  font-size: 14px;
+  min-width: 200px;
+
+  &:focus {
+    outline: none;
+    border-color: rgba(255, 255, 255, 0.2);
```

Comment:
I know visuals don't really matter as much in admin tools but it would be nice to use the common CSS variable stuff, I have a todo thing to clean up all the places we do `rgba(255, 255, 255, ###)` because it looks much worse than using a specific shade of blue/grey-blue, and it would be nice to not have to come clean up this file that you just created